wasm/names.go

```diff
@@ -0,0 +1,74 @@
+package wasm
+
+import (
+	"github.com/tetratelabs/wazero/wasm/leb128"
+)
+
+// nameSectionPrefix a length-prefixed 'name'. subsection data should follow.
+var nameSectionPrefix = encodeName("name")
+
+// encodeCustomNameSection encodes a possibly empty buffer representing the "name" wasm.Module CustomSection.
+// See https://www.w3.org/TR/wasm-core-1/#name-section%E2%91%A0
+func encodeCustomNameSection(m *Module) []byte {
```

Comment:
Given that `*Module` already contains all the custom sections in module.CustomSections and possibly original `"name"` section there, what's the usecase for this?

Reply:
here are the use cases:
 * round tripping test not requiring hex editor and reviewers to know exact encoding rules
 * less performance lost wat to wasm
 * more intuitive debugging as the name serialized by the compiler ex even tinygo is visible

we can address these later, though, so resolving!